server/etcdmain/config.go

```diff
@@ -160,6 +160,7 @@ func newConfig() *config {
 	fs.DurationVar(&cfg.ec.GRPCKeepAliveMinTime, "grpc-keepalive-min-time", cfg.ec.GRPCKeepAliveMinTime, "Minimum interval duration that a client should wait before pinging server.")
 	fs.DurationVar(&cfg.ec.GRPCKeepAliveInterval, "grpc-keepalive-interval", cfg.ec.GRPCKeepAliveInterval, "Frequency duration of server-to-client ping to check if a connection is alive (0 to disable).")
 	fs.DurationVar(&cfg.ec.GRPCKeepAliveTimeout, "grpc-keepalive-timeout", cfg.ec.GRPCKeepAliveTimeout, "Additional duration of wait before closing a non-responsive connection (0 to disable).")
+	fs.StringVar(&cfg.ec.ReadOnlyOption, "read-only-option", cfg.ec.ReadOnlyOption, "Option to support the linearizability of the read only request(ReadOnlySafe and ReadOnlyLeaseBased are supported options).")
```

Comment:
I wonder whether we should expose this as --experimental first 
(or we have evidence of running with this mode in production ?)

Maybe we should name this file more meaningful,e.g. ? 
--experimental-linearizable-reads-mode

"Read-only-option" I would interpret as running etcd as a RO service, so guarantying immutability.

Reply:
Yeah, this is a good suggestion, I agree we should make this flag as experimental first, as this mode is rarely used in the production environment at present. I will fix this later.